java/src/jmri/NamedBean.java

```diff
@@ -289,4 +289,22 @@
     @CheckReturnValue
     @Nonnull
     public String getBeanType();
+    
+    /**
+     * Enforces, and as a user convenience converts to, the standard form for a user name.
+     * <p>
+     * This implementation just passes the name through, but later versions might 
+     * e.g. trim leading and trailing spaces.
+     *
+     * @param inputName 
+     * @throws CantNormalizeNameException If the inputName can't be converted to normalized form
+     * @return A user name in standard normalized form 
+     */
+    @CheckReturnValue
+    static public @Nonnull String normalizeUserName(@Nonnull String inputName) throws BadUserNameException {
+        // uncomment next line to allow debugging of trimmed user names
+        // return inputName.trim();
+        return inputName;
+    }
+    public class BadUserNameException extends IllegalArgumentException {}
```

Comment:
Could be.  There are a bunch of both Bad* and Invalid* subclasses of Exception [defined by Java](https://docs.oracle.com/javase/8/docs/api/java/lang/Exception.html).  I don't really see a pattern to when one is used vs the other. I'd recently run across `BadLocationException` and `BadStringOperationException` on another project, so took those as a model.